UserGrid Upgrades

manchuck opened the pull request:
Added Check for token expiration before trying to call user grid
Added default time out for making requests so python can time out
Created test cases
Removed un-unused get_actor function
Deprecated print_user
Adding logging facality
Added some helpful proxy functions:i
  - get_entity_by_id
  - delete_entitity_by_id
  - put_entity_by_id

Restricting access to some functions

manchuck committed:
Adding unit tests

manchuck committed:
UserGrid Upgrades

Added Check for token expiration before trying to call user grid
Added default time out for making requests so python can time out
Created test cases
Removed un-unused get_actor function
Deprecated print_user
Adding logging facality
Added some helpful proxy functions:i
  - get_entity_by_id
  - delete_entitity_by_id
  - put_entity_by_id

Restricting access to some functions

manchuck committed:
Added Linter along with lint fixes for usergrid

smitster commented on setup.py:
```diff
@@ -1,14 +1,17 @@
-    include_package_data = True
+    packages=find_packages(exclude=['tests']),
+    package_dir={
+        'usergird': 'usergrid'
```
typo - 'usergird'

smitster commented on usergrid/usergrid.py:
```diff
@@ -1,450 +1,610 @@
-            print("UG_client:" + str(txt))
+import requests
+
+__version__ = '0.1.3'
```
same version bump as setup.py

smitster commented on usergrid/usergrid.py:
```diff
@@ -1,450 +1,610 @@
+        cursor = None
+
+        if not limit or limit > 100:
+            limit = 100
```
collect_entities will page through as long as it has a cursor. There's definitely an advantage to doing this at the UG maximum, 1000 - fewer but larger responses is faster than many smaller responses in adhoc testing over the last 2 years.

smitster reviewed (changes requested):
Mostly looks good. A typo ('usergird'), version bump, and for collect_entities I think we leave it at 1000 (UG max limit) - that has performed better for me than making more requests with smaller responses.

manchuck committed:
Updated collect_entities to page at a 1000 and fixed a small issue with the generator
Updated version number to be correct version in all files
Fixed typeo in setup.py
Added logger to mock_usergrid and removed print statements
Disabled logger by default in tests/__init__.py

manchuck commented:
@smitster I added some tests for collect_entities and process entities and bumped it back up to 1000 however Im not going to mock a request that returns 1000 entities in a response.  

I changed it to 100 and a generator instead because I noticed that @shelatpinal was using it in the path-service to do some remove that path from all stories during deletion.  Since that is a long running process I wanted to curve garbage collection issues.

smitster commented:
Great. That's fair about the mock response. It should have been a generator to begin with, so that's great, but we'll want to keep our eyes out for where and how collect_entities is used - there are some pieces of code that get len() of the results, and should also give us some sense of whether the response times from n=1000 merit the additional potential gc pressure

nojyarg reviewed (approved):
LGTM
